### PR TITLE
#1123 - Concrete intersection between half-space and polytopic reset map

### LINKED ISSUE #1123: Concrete intersection between half-space and polytopic reset map
Wanted: `intersection(::HalfSpace{N}, ::LazySets.ResetMap{N, <:AbstractPolytope})`

This would work:
```julia
function intersection(hs::HalfSpace{N}, rm::ResetMap{N, <:AbstractPolytope{N}}) where {N<:Real}
    return HPolytope([rs; constraints_list(rm)])
end
```

### PR DESCRIPTION
**Pull request by schillic**
Closes #1123.

### CODE REVIEW

File: src/concrete_intersection.jl [hunk 1]
```diff
@@ -550,3 +550,31 @@ end
...
+
+An `HPolytope`.
+
+### Notes
+
+We assume that `rm` is polytopic, i.e., is bounded and has a `constraints_list`
+method defined.
+"""
+function intersection(P::AbstractPolyhedron{N}, rm::ResetMap{N}) where {N<:Real}
+    return HPolytope([constraints_list(P); constraints_list(rm)])
```
Comment (mforets) [inline src/concrete_intersection.jl]: Have you considered removing redundant constraints? This is done by default for polytope intersections [here](https://github.com/JuliaReach/LazySets.jl/blob/master/src/concrete_intersection.jl#L340) and i would suggest to use it here as well.

Comment (mforets) [reply src/concrete_intersection.jl]: Perhaps we can just invoke that function but doing `HPolytope(constraints_list(rm))`?

Comment (schillic) [reply src/concrete_intersection.jl]: Yes, I considered it, but decided against it.
I changed the code (it was actually unsound) and now it uses the method you linked to.

Comment (mforets) [reply src/concrete_intersection.jl]: Great, looks good to me. The argument behind removing redundant constraints by deafult is that is simplifies a lot of cases where the intersection is actually empty. And, well if the intersection is not empty you get a simpler polyhedron. But we can add a flag to turn this on or off, currently there is no such flag.

**Pull request merged**